src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -379,13 +379,16 @@ private int countCurrentEC2Slaves(SlaveTemplate template) throws AmazonClientExc
         Set<String> instanceIds = new HashSet<String>();
         String description = template != null ? template.description : null;
 
+        //FIXME convert to a filter query
```

Comment:
:fireworks: :+1: these growing loops and ifs are becoming more and more scary :-).

Reply:
Yes ... we have to replace with a query on the AWS side, that will reduce the data transfer and the complexity of the if, I promise will be in the 1.43 :-)